jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -637,7 +637,9 @@ def calculate_photometry(self, dataset=None, aperture=None, background=None,
                     include_pixarea_fac = True
             if img_unit != u.count:
                 try:
-                    ctfac = float(counts_factor if counts_factor is not None else self.counts_factor)  # noqa
+                    ctfac = float(counts_factor if counts_factor is not None else self.counts_factor)  # noqa: E501
+                    if ctfac < 0:
+                        raise ValueError('Counts conversion factor cannot be negative.')

```

Comment:
is this by design that the user will actually see the other error (I see the test does test for that, so I'm assuming its intended, just want to double check)?

Reply:
The design is so `except` catches it. The message does not propagate to user based on current logic. I thought about modifying that logic but that seems out of scope.